### PR TITLE
[snapshots] populate snapshot details

### PR DESCRIPTION
**Pull request by sharder996**
Rebased with changes from #3136 to make testing easier

### CODE REVIEW

File: src/daemon/daemon.cpp
Comment (ricab) [inline src/daemon/daemon.cpp:1638]: This function is getting really big, with all the sub functions and all. I think this would be a good time to extract the individual parts into either private methods or unnamed namespace helpers.

**Review by ricab — CHANGES_REQUESTED** [review]
Hey @sharder996, thanks for this work! This is only a superficial review, mainly to adjust the approach a little, but this seems to be working nicely.

File: src/daemon/daemon.cpp
Comment (ricab) [inline src/daemon/daemon.cpp:1720]: I don't think you need this here, since you are iterating over the individual snapshot arguments in the next few lines.

File: src/daemon/daemon.cpp
Comment (ricab) [inline src/daemon/daemon.cpp:1715]: This is now clearly a misnomer. Can we change it to `all_or_none`, here and in the `SnapshotPick`? Or something better you can think of.

File: src/daemon/daemon.cpp
Comment (ricab) [inline src/daemon/daemon.cpp:3397]: `runtime_info` is the opposite of what this parameter currently means, isn't it? 

I agree it would make for sense to drop the negative, but then the logic needs inverting too, i.e. drop the `!` here and add it where you pass the argument from the grpc request.

File: src/daemon/daemon.cpp
Comment (ricab) [inline src/daemon/daemon.cpp:1765]: Why didn't you include these lines in `populate_snapshot_fundamentals`, instead of repeating them?

**Review by ricab — COMMENTED** [review]
Looking great! Thanks a lot for reworking this, it is much better :slightly_smiling_face: I have some more requests, but they should all be easy.

File: src/daemon/daemon.cpp
Comment (sharder996) [reply src/daemon/daemon.cpp:1761]: I wanted to keep `populate_snapshot_fundamentals` aligned with the message type `SnapshotFundamentals` and not change anything outside of it. There is also [one situation](https://github.com/canonical/multipass/blob/ac8a96cb4d9819136f68f428151cdeb0f1359154/src/daemon/daemon.cpp#L1327) where the fundamentals are set and not `instance_name`.

Comment (ricab) [reply src/daemon/daemon.cpp:1761]: I see. But then you can extract a helper employ here, similar to what happened already with `get_snapshot_info`.

**Review by ricab — APPROVED** [review]
Alright, LGTM!

bors r+

**Comment by ricab** [comment]
duh, no need for bors on this one

bors cancel

**Comment by ricab** [comment]
I'll just wait for CI before merging.

**Pull request merged**
